src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -0,0 +1,57 @@
+import {useEffect} from 'react';
+import type {ComponentType} from 'react';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import useNetwork from '@hooks/useNetwork';
+import {openPolicyAccountingPage} from '@libs/actions/PolicyConnections';
+import withPolicy from './withPolicy';
+import type {WithPolicyProps} from './withPolicy';
+
+type WithPolicyConnectionsProps = WithPolicyProps;
+
+function withPolicyConnections(WrappedComponent: ComponentType<WithPolicyConnectionsProps>) {
+    function WithPolicyConnections({policy, policyMembers, policyDraft, policyMembersDraft, route}: WithPolicyConnectionsProps) {
+        const {isOffline} = useNetwork();
+
+        useEffect(() => {
+            if (!policy?.areConnectionsEnabled || !!policy?.connections || !policy?.id) {
+                return;
+            }
+
+            // If the policy doesn't have the connections data when it has the accounting connections feature enabled,
+            // call the API to get the data
+            openPolicyAccountingPage(policy.id);
```

Comment:
@aldo-expensify, is it accurate to say that the `connections` object is empty when the user hasn't connected to any accounting software?

If that is the case, this `useEffect` will keep calling `openPolicyAccountingPage` indefinitely for workspaces that have the accounting feature enabled but that haven't connected to any accounting software.

Do you have any suggestion on how we can distinguish the situation when the user has connected to the accounting software but the `connections` object hasn't populated? This is the only situation when we have to call `openPolicyAccountingPage`.

Reply:
> @aldo-expensify, is it accurate to say that the connections object is empty when the user hasn't connected to any accounting software?

I just tested and it appears to me that the `connections` key is completely missing. I guess what you could do is create a local state with `useState` so you are sure that you only call `openPolicyAccountingPage` once. Something like

```
if (!loadedPolicyConnections) {
   openPolicyAccountingPage();
   setLoadedPolicyConnections(true);
}
```

Reply:
Aldo's plan would work, but it might result in more API requests than necessary.

An alternate thought I wad was that maybe in the policy summaries that get sent to NewDot we could include a new property like `connectCount: 2` so that we know how many connections to expect. That way, you could determine the difference between not having any connections configured, and not having the connections loaded in Onyx yet.

Reply:
Thank you for the thoughtful and creative ideas, both 😄 

For now, I'll go with @aldo-expensify's suggestion as it's easier to implement. As an optimization to mitigate unnecessary API calls, I'll create a GH issue to implement @tgolen's suggestion.

Reply:
Thanks for addressing it. If there isn't any real deadline for this, why not just do it the correct way the first time?

Reply:
Tim, thank you for the question. My thinking is that your solution, although the most preferable one, requires additional PRs where we add codes to create, increment, and decrement the new counter field.

Although QBO project itself doesn't have an external commit, the Xero project that comes after QBO has one. That's why I wanted to prioritize the speed.

However, let me know if my thinking is not the best course of action we can take. I'm happy to adjust our plans if needed 😄

Reply:
Yeah, that all makes sense. I think it's important to do it right the first, or else you're also having to do the work to remove the initial solution.

Even with Xerocon in August, I don't think it would take more than a day or two in order to add the property to the policy summaries. Do you think it would be much more than that? I was just thinking that when the policy list is fetched, it would also query the number of connections, so we don't have to worry about incrementing or decrementing a counter.

Reply:
I'll start working on https://github.com/Expensify/Expensify/issues/383660 to add the counter field to the `policy` object

Reply:
@mollfpr, let's get this merged

Reply:
> @aldo-expensify, if I remember correctly, the `ConnectToQuickbooksOnline` Web-E command is called after the QBO authentication is done and it handles pushing the Onyx update for the Policy object to the clients?

`ConnectToQuickbooksOnline` is called to start the authentication flow, so before... `ConnectToQuickbooksOnline` itself doesn't send any onyx updates. The job in the IS server does and when we call `SavePolicy` further down the flow.

I don't think in any case we send the full policy object. I think the updates only contain the updated sections handled by SavePolicy.

Reply:
I'm a bit concerned about the `connectCount` making things complicated unnecessarily. We would have to make sure we are always updating the `connectCount` when there is a `connections` change. 

What about instead of using a local state as I suggested, we use a context with a state, so if we open more pages that require the policy's connection settings, we can see that a different page already loaded it.

Reply:
> What about instead of using a local state as I suggested, we use a context with a state, so if we open more pages that require the policy's connection settings, we can see that a different page already loaded it.

This is a sound idea. It addresses @tgolen's concerns about unnecessary API calls to a certain extent.

@tgolen, do you have any other concerns besides unnecessary API calls? I'm interested in understanding the potential issues your solution aims to resolve. 😄

Reply:
My main concerns are unnecessary API calls, yes. They do not have a zero cost and I think it is a very naive approach there is a better solution. Whatever you do here is going to put a pattern in place that engineers are going to follow from here on out and I think it's a bad pattern.

It sounds like @aldo-expensify is concerned that a piece of policy data would be complex to keep track of, but this is just like any other piece of policy that exists on the policy and there is nothing more complex that needs to be done here that we aren't doing with all of the other policy data.

Reply:
I think we should seek more opinions in engineering-chat, but meanwhile... I don't think the problem of "unnecessary API calls" is big enough to introduce this new piece of data. Why? The only cases that would happen are:
- User navigates to the main Accounting page: if `policy.connections` is missing, it will trigger a fetch, but just one and will get nothing (there was no `connections`).
- User navigates to deeper Accounting page when there is no `policy.connections`: this can only happen if the user is entering directly using the url, which I think is a rare case.

In cases where there is actually a connection, after we load it, we won't do further calls to the API

Reply:
Slack discussion: https://expensify.slack.com/archives/C03TQ48KC/p1712185841746469

Reply:
We came to the consensus that we should go with the approach to fetch the `connections` data once.

I'll modify the PR to use a Context as suggested by @aldo-expensify 👍